components/dialog/dialog-mixin.js

```diff
@@ -44,6 +44,7 @@ export const DialogMixin = superclass => class extends RtlMixin(superclass) {
 			_overflowBottom: { type: Boolean },
 			_overflowTop: { type: Boolean },
 			_parentDialog: { type: Object },
+			_scroll: { type: Boolean },
```

Comment:
Yeah the browser is scrolling on its own, even though the input may be close to the top.  I was able to reproduce the issue in the demo page by just adding 20-30 input elements, but using `scrollIntoView(false)` as originally intended has a couple of issues.  

First is that `_focusFirst` gets called whenever focus enters the trap, so if the user opens the dialog, moves focus outside the document (ex. to the address bar), and then tabs back into the document, then it will get called again and will cause the dialog to scroll to the top even if it doesn't need to.  This is a minor edge case but it leads me to think that this behaviour should be tied to the open state change instead.  

The second issue which prompts the use of this new private `_scroll` variable is that the user can see the scroll action (on MacOS the scrollbar flashes briefly upon opening the dialog). The solution to that is to hide the overflow until after the scroll has been updated.  This could actually be done inside of `_focusFirst` instead, if it were not for the first issue.

The timing of this is also apparently sensitive.  I had to make sure to do this before the size logic runs, otherwise FF shows the scroll shadow on top even though it shouldn't - although I'm not entire sure what's going on there, or why it's different from the rest of the browsers.

Reply:
Amazing, makes sense. I still have a TODO to come back to the fact that all this focus stuff is broken in Safari and that whole discussion you and I had about how table overrides the focus-trap stuff just to avoid the close button. Good times.